### ISSUE TITLE
apollo 5.5 & 6.0 module/planning/tasks/deciders: LaneChangeDecider::IsPerceptionBlocked() question

### ISSUE THREAD

**Issue by zynlemon** [ApolloAuto/apollo#13882]
We appreciate you go through Apollo documentations and search previous issues before creating an new one. If neither of the sources helped you with your issues, please report the issue using the following form. Please note missing info can delay the response time.

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 18.04)**: Linux Ubuntu 20.04
- **Apollo installed from (source or binary)**: official git reporsitory
- **Apollo version (3.5, 5.0, 5.5, 6.0)**: version 5.5 & 6.0
- **Output of `apollo.sh config` if on `master` branch**: don't know what this mean

### Steps to reproduce the issue:
- **Please use bullet points and include as much details as possible**:
hello dear developers, I have a question when reading planning module code:
in LaneChangeDecider::IsPerceptionBlocked():
```
right_most_angle = beam_heading
...
left_most_angle = beam_heading - search_angle
```
then:
```
std::fabs(common::math::NormalizeAngle(left_most_angle - right_most_angle))
```
is this angle the search_angle when beam first blocked by obstacle poligon point?
what is the meaning of this angle?
should left_most_angle = beam_heading and then the angle mentioned before becomes the angle that beam blocked by the obstacle?

### Supporting materials (screenshots, command lines, code/script snippets):
![IsPerceptionBlocked](https://user-images.githubusercontent.com/76237917/123089565-afb53b80-d459-11eb-8e07-923ce0f70722.png)

**Comment by daohu527** [comment]
The beam_heading is rotate from right to left, and when it found obstacle then will search for the obstacle left boundary. If the (left-right) > 0.5(radian) then means it is blocked. 

## sorted
`reference_line_info.path_decision().obstacles()` should be sort from right to left, Otherwise `right_most_angle` will be wrong

#### right_most_angle 
if obstacles are sorted, then `right_most_angle` will found only once be **the right boundary of the rightmost obstacle**.

#### left_most_angle
after found `right_most_angle`, `left_most_angle` will continue found be `0`

## In summary
So, this is an issue to be fixed, thank you for your submission.

The real intention is shown in the figure below
![Agile Board](https://user-images.githubusercontent.com/10419854/124408122-20bcf300-dd78-11eb-8c52-16e653595fd9.jpg)

**Comment by tuanngokien** [comment]
Hi @daohu527,
I want to ask about the formula in function`IsPerceptionBlocked`. 
In fact, I'm fine with your XY-plane figure visualizing the `left_most_angle` and `right_most_angle`. But when I see in the actual code (after @randomacco's fixed commit), I still cannot figure out how dev calculated `left_most_angle`.

```
const double beam_heading = common::math::NormalizeAngle(
          adv_heading - 0.5 * search_range + search_angle);
...

left_most_angle = beam_heading - search_angle;
```

And this infers `left_most_angle = adv_heading - 0.5 * search_range` which is a difference of 2 constant values, and consequently the result of `left_most_angle` is constant and does not depend on `search_angle`. 

How this could be correct because `left_most_angle` and `right_most_angle` need to vary dynamically based on size of specific obstacle.

**Comment by daohu527** [comment]
Yes I think it needs to be added in addition to fixing here
```
      if (right_most_found && !obstacle_polygon.HasOverlap(search_beam)) {
        left_most_angle = beam_heading;
        break;
      }
```

**Comment by tuanngokien** [comment]
> Yes I think it needs to be added in addition to fixing here
> 
> ```
>       if (right_most_found && !obstacle_polygon.HasOverlap(search_beam)) {
>         left_most_angle = beam_heading;
>         break;
>       }
> ```

Thank for your answer. I've thought the same as you, and cannot explain why they calculated differently in the actual code.
